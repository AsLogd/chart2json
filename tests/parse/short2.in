[Song]
{
  Name = "Song name"
  Artist = "Artist"
  Year = ", 2016"
  Resolution = 192
  Player2 = bass
  Genre = "Rock"
  MediaType = "cd"
  MusicStream = "song.ogg"
}
[SyncTrack]
{
  0 = TS 4
  0 = B 100000
  1536 = B 120000
  1536 = A 123456
}
[Events]
{
  0 = E "section intro"
  30000 = E "section chorus"
  40000 = E "section solo"
}
[ExpertSingle]
{
  1000 = N 1 1000
  1500 = S 2 3000
  1500 = N 2 0
  1500 = N 3 0
  1800 = N 3 0
  1800 = N 5 0
  2000 = N 4 0
  2000 = N 6 0
}
