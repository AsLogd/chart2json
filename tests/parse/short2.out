{
    "song": {
        "resolution": 192,
        "audioStreams": {
            "MusicStream": "song.ogg"
        },
        "name": "Song name",
        "artist": "Artist",
        "year": 2016,
        "player2": "bass",
        "genre": "Rock",
        "mediaType": "cd"
    },
    "syncTrack": {
        "0": {
            "events": [
                {
                    "kind": 1,
                    "signature": {
                        "numerator": 4,
                        "denominator": 4
                    }
                },
                {
                    "kind": 0,
                    "bpm": 100
                }
            ]
        },
        "1536": {
            "events": [
                {
                    "kind": 0,
                    "bpm": 120,
                    "anchorMicroSeconds": 123456
                }
            ]
        }
    },
    "difficulties": {
        "Expert": {
            "Single": {
                "1000": {
                    "events": [
                        {
                            "lanes": [
                                {
                                    "lane": 2,
                                    "sustain": 1000
                                }
                            ],
                            "forced": false,
                            "tap": false
                        }
                    ]
                },
                "1500": {
                    "events": [
                        {
                            "lanes": [
                                {
                                    "lane": 3,
                                    "sustain": 0
                                },
                                {
                                    "lane": 4,
                                    "sustain": 0
                                }
                            ],
                            "forced": false,
                            "tap": false
                        },
                        {
                            "type": 2,
                            "duration": 3000
                        }
                    ]
                },
                "1800": {
                    "events": [
                        {
                            "lanes": [
                                {
                                    "lane": 4,
                                    "sustain": 0
                                }
                            ],
                            "forced": true,
                            "tap": false
                        }
                    ]
                },
                "2000": {
                    "events": [
                        {
                            "lanes": [
                                {
                                    "lane": 5,
                                    "sustain": 0
                                }
                            ],
                            "forced": false,
                            "tap": true
                        }
                    ]
                }
            }
        }
    },
    "events": {
        "0": {
            "events": [
                {
                    "kind": 0,
                    "name": "intro"
                }
            ]
        },
        "30000": {
            "events": [
                {
                    "kind": 0,
                    "name": "chorus"
                }
            ]
        },
        "40000": {
            "events": [
                {
                    "kind": 0,
                    "name": "solo"
                }
            ]
        }
    }
}